apm-agent-plugins/apm-spring-webmvc-plugin/src/test/java/co/elastic/apm/agent/spring/webmvc/ExceptionHandlerInstrumentationWithExceptionResolverTest.java

```diff
@@ -0,0 +1,54 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 - 2019 Elastic and contributors
+ * %%
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.spring.webmvc;
+
+import co.elastic.apm.agent.spring.webmvc.testapp.exception_resolver.ExceptionResolverController;
+import co.elastic.apm.agent.spring.webmvc.testapp.exception_resolver.ExceptionResolverRuntimeException;
+import co.elastic.apm.agent.spring.webmvc.testapp.exception_resolver.RestResponseStatusExceptionResolver;
+import org.junit.Test;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.ResultActions;
+
+import static org.junit.Assert.assertEquals;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+
+@ContextConfiguration(classes = {
+    ExceptionResolverController.class,
+    RestResponseStatusExceptionResolver.class})
+public class ExceptionHandlerInstrumentationWithExceptionResolverTest extends AbstractExceptionHandlerInstrumentationTest {
+
+    @Test
+    public void testCallApiWithExceptionThrown() throws Exception {
+        ResultActions resultActions = this.mockMvc.perform(get("/exception-resolver/throw-exception"));
+        MvcResult result = resultActions.andReturn();
+        MockHttpServletResponse response = result.getResponse();
+
+        assertExceptionCapture(0, 1, ExceptionResolverRuntimeException.class, response, 200, "", "runtime exception occured");
```

Comment:
Why don't we get a span now? Was it the `DispatcherServlet#render` span before?
And why do we capture this Exception now even if I disable the new instrumentation? Is it thrown out now and not handled?

Reply:
Yes, it was DispatcherServlet#render span.
With disabled instrumentation of DispatcherServlet#render attribute "name org.springframework.web.servlet.DispatcherServlet.EXCEPTION" is exist in request headers when testing ExceptionHandlerInstrumentationWithExceptionResolverTest.

Reply:
Earliear we added captureException in DispatcherServlet#render instrumentation. Because of this we have 2 errors.
Currenly, without ExceptionHandlerInstrumentation in HttpServletRequest  attribute "org.springframework.web.servlet.DispatcherServlet.EXCEPTION" is present

Reply:
> With disabled instrumentation of DispatcherServlet#render attribute "name org.springframework.web.servlet.DispatcherServlet.EXCEPTION" is exist in request headers when testing ExceptionHandlerInstrumentationWithExceptionResolverTest

Not sure I get this right- does the instrumentation change the attributes stored in the request?? Did you verify that? If that's the case, we need to see why. We shouldn't affect request attributes not set by us in any case. 

Can you say what had changed between the commit where `ExceptionHandlerInstrumentationWithExceptionResolverTest` yield no error and now? As it stands now, nothing tests `ExceptionHandlerInstrumentation`, so we can't merge this way.

Let us know if you want us to take over.
Thanks a lot for your efforts! This is highly appreciated!

Reply:
> > With disabled instrumentation of DispatcherServlet#render attribute "name org.springframework.web.servlet.DispatcherServlet.EXCEPTION" is exist in request headers when testing ExceptionHandlerInstrumentationWithExceptionResolverTest
> 
> Not sure I get this right- does the instrumentation change the attributes stored in the request?? Did you verify that? If that's the case, we need to see why. We shouldn't affect request attributes not set by us in any case.
> 
> Can you say what had changed between the commit where `ExceptionHandlerInstrumentationWithExceptionResolverTest` yield no error and now? As it stands now, nothing tests `ExceptionHandlerInstrumentation`, so we can't merge this way.
> 
> Let us know if you want us to take over.
> Thanks a lot for your efforts! This is highly appreciated!

I should delete this comment. The DispatcherServlet#render instrumentation not change the attributes. Earlier, that test failed, but now it work correct. 
In the end, we don't need in ExceptionHandlerInstrumentation. 
If it need, I can try to test that commit version to see what was that problem.

Reply:
If you can, that would be great.
Take a look at [the code I referred above](https://github.com/elastic/apm-agent-java/pull/759#discussion_r312891657)- it is expected that when Exception is handled quietly like you did, neither attribute will be set when exiting the Servlet APIs `doFilter` and `service`.

Reply:
@eyalkoren , I found the reason :)
I added manual attribute and forgot delete it.
```
@Override
    protected ModelAndView doResolveException(HttpServletRequest request, HttpServletResponse response, Object handler, Exception e) {
        ModelAndView model = new ModelAndView("error-page");
        model.addObject("message", "runtime exception occured");
        request.setAttribute(DispatcherServlet.EXCEPTION_ATTRIBUTE, e);
        return model;
    }
```
I will close this merge request, and recreate new.